posthog/src/main/java/com/posthog/internal/PostHogQueue.kt

```diff
@@ -315,3 +315,26 @@ internal class PostHogQueue(
             return tempFiles
         }
 }
+
+private fun calcFloor(currentValue: Int): Int {
+    return max(currentValue.floorDiv(2), 1)
+}
+
+internal fun deleteFilesIfAPIError(
+    e: PostHogApiError,
+    config: PostHogConfig,
+): Boolean {
+    if (e.statusCode < 400) {
+        return false
+    }
+    // workaround due to png images exceed our max. limit in kafka
+    if (e.statusCode == 413 && config.maxBatchSize > 1) {
+        // try to reduce the batch size and flushAt until its 1
+        // and if it still throws 413 in the next retry, delete the files since we cannot handle anyway
+        config.maxBatchSize = calcFloor(config.maxBatchSize)
+        config.flushAt = calcFloor(config.flushAt)
```

Comment:
nice... it makes me wonder how we'd know it was happening 🤔 but I guess we can figure out where we're returning 413 in capture and send an ingestion warning

Reply:
yeah we probably have 413s in our logging systems, adding one more custom event to the batch is likely to increase the payload :(